test/e2e/framework/service/jig.go

```diff
@@ -829,6 +829,18 @@ func testReachabilityOverClusterIP(clusterIP string, sp v1.ServicePort, execPod
 	return nil
 }
 
+func testReachabilityOverExternalIP(externalIP string, sp v1.ServicePort, execPod *v1.Pod) error {
+	// If .spec.externalIPs[]externalIP is set to "" or "None", reachability can not be tested over externalIP:servicePort
+	isClusterIPV46, err := regexp.MatchString(e2enetwork.RegexIPv4+"||"+e2enetwork.RegexIPv6, externalIP)
+	if err != nil {
+		return fmt.Errorf("unable to parse ClusterIP: %s", externalIP)
+	}
+	if isClusterIPV46 {
+		return testEndpointReachability(externalIP, sp.Port, sp.Protocol, execPod)
+	}
+	return nil
+}
```

Comment:
```suggestion
func testReachabilityOverExternalIP(externalIP string, sp v1.ServicePort, execPod *v1.Pod) error {
	// If .spec.externalIPs[]externalIP is set to "" or "None", reachability can not be tested over externalIP:servicePort
```suggestion
	ip := net.ParseIP(externalIP)
	if ip == nil {
		return fmt.Errorf("unable to parse ClusterIP: %s", externalIP)
	}
	return testEndpointReachability(externalIP, sp.Port, sp.Protocol, execPod)

```

Reply:
Thanks, I've added these changes as well as fixing the error message to say ExternalIP instead of ClusterIP